.github/workflows/lib-tests.yml

```diff
@@ -0,0 +1,69 @@
+name: pytests
+on: [push]
+jobs:
+  pytest:
+    runs-on: self-hosted
+
+    steps:
+
+      - name: nvidia-smi test
+        run: nvidia-smi
+
+      - name: pycuda test
+        run: python3 -c "import pycuda.autoinit; print('yes')"
+      
+      - uses: actions/checkout@v1
+      
+      - name: upgrade pip
+        run: python3 -m pip install --upgrade pip
+
+      - name: put ~/.local/bin on $PATH
+        run: echo "PATH=$HOME/.local/bin:$PATH" >> $GITHUB_ENV
+
+      - name: create a requirements file
+        run: echo "numba scipy numpy networkx scikit-learn cython numexpr psutil pytest pytest-cov pytest-xdist pytest-benchmark pytest-mock matplotlib psycopg2 h5py>=2.10 typing BeautifulSoup4 subprocess32 flask-restx python-keycloak mako pyAesCrypt pyunicore formencode cfflib jinja2 nibabel sqlalchemy alembic allensdk sphinx==1.2.3 docutils==0.12 werkzeug flask gevent jupyter cherrypy autopep8" | tr ' ' \\n > requirements.txt
+
+      - name: cache ~/.local for pip deps
+        id: cache-local
+        uses: actions/cache@v2
+        with:
+          path: ~/.local
+          key: ${{ runner.os }}-pip-${{ hashFiles('requirements.txt') }}
+
+      - name: pip install dependencies (from Dockerfile-build)
```

Comment:
eh, indeed, it's duplicate, but the idea is that maybe a PR wants/needs to change the dependency version.  Having the  standard set installed in the Dockerfile means that if you want to upgrade matplotlib, it's a short process and not re-downloading all the packages for each CI run.  Since this is running at our faculty, bandwidth is good but not the same as The Cloud.